src/managers/resources/emg.rs

```diff
@@ -29,16 +81,82 @@ impl Resource for Emg {
 
 impl ResourceManager for Manager<Emg> {
     type ResourceType = Emg;
-    /// Handles all EMG-related tasks
+
     async fn handle_task(&mut self, channel_data: ManagerChannelData) -> Result<()> {
-        let (task, _, send_channel) =
+        let (task, task_data, send_channel) =
             parse_channel_data!(channel_data, Task, EmgData).map_err(|e: Error| e)?;
```

Comment:
```suggestion
            parse_channel_data!(channel_data, Task, EmgData)?;
```
Ah, sorry, the redundant `map_err` here is my bad -- could you also make this change in the other resource files. Thank you!